### DISCUSSION TITLE
Some settings to load automatically API keys

### DISCUSSION THREAD

**Discussion by Louvivien** [FlowiseAI/Flowise#79] [Ideas]
- .env compatibility
- Settings somewhere in the front

**Comment by HenryHengZJ** [comment]
mind to clarify abit more about how to load api keys automatically?

**Reply by matbeedotcom** [reply]
dotenv

**Comment by Louvivien** [comment]
There would be two possibilities: 
- for non developer there would be a settings feature where you could set up your api keys. When entered once if you load a file with existing components or add a component that you have already used it will populate the Keys automatically. Additionally if there are several OpenAi components or Pinecone components, you would only need to populate one and the other one would be automatically added. 
- for developers locally if you would have a .env file it would populate automatically the settings (when deployed it would be compatible with environment variables)
This needs to be secure obviously

**Reply by HenryHengZJ** [reply]
This is really good input! Save you from entering api keys repeatedly. Will add that to feature list
